async_service/trio.py

```diff
@@ -156,7 +162,19 @@ def cancel(self) -> None:
         *args: Any,
         daemon: bool,
         name: str,
+        parent,
+        task_status=trio.TASK_STATUS_IGNORED,
     ) -> None:
+        task_status.started()
+        current_task = trio.hazmat.current_task()
+        self.logger.info("Starting task %s, child of %s", current_task, parent)
+        async with self._task_nurseries_lock:
+            # We need to pre-populate self._task_nurseries with a nursery for this task so that if
+            # it spawns any sub-tasks, we know which nursery to run it from.
+            nursery = self._task_nurseries.get(current_task)
+            if nursery is None:
+                nursery = await trio.open_nursery().__aenter__()
```

Comment:
I think I tried this once and it didn't work out very well.  I'm thinking that *maybe* this will have to be done somewhere like the internal `_run_and_manager_task()` since it lives for the duration of the task and to actually use the context manager API to open the nursery.

Reply:
Well, this is inside `_run_and_manage_task()`. Or maybe you mean something else?